src/components/MultiHopTrade/hooks/useThorStreamingProgress/useThorStreamingProgress.tsx

```diff
@@ -0,0 +1,130 @@
+import type { ProgressProps } from '@chakra-ui/progress'
+import axios from 'axios'
+import { getConfig } from 'config'
+import { useEffect, useMemo, useRef, useState } from 'react'
+import { usePoll } from 'hooks/usePoll/usePoll'
+const POLL_INTERVAL_MILLISECONDS = 30_000 // 30 seconds
+
+type ThornodeStreamingSwapResponse = {
+  tx_id?: string
+  interval?: number
+  quantity?: number
+  count?: number
+  last_height?: number
+  trade_target?: string
+  deposit?: string
+  in?: string
+  out?: string
+  failed_swaps?: number[]
+  failed_swap_reasons?: string[]
+}
+
+export const getThorchainStreamingSwap = async (
+  sellTxHash: string,
+): Promise<ThornodeStreamingSwapResponse | undefined> => {
+  const thorTxHash = sellTxHash.replace(/^0x/, '')
+  const { data: streamingSwapData } = await axios.get<ThornodeStreamingSwapResponse>(
+    `${getConfig().REACT_APP_THORCHAIN_NODE_URL}/lcd/thorchain/swap/streaming/${thorTxHash}`,
+  )
+
+  if (!streamingSwapData) return
```

Comment:
And handle the errored response here
```suggestion
  if (!streamingSwapData || streamingSwapData.error) return
```